metaflow/plugins/aws/batch/batch.py

```diff
@@ -248,6 +257,9 @@ def create_job(
                 AWS_SECRETS_MANAGER_DEFAULT_REGION,
             )
 
+        if tmpfs_tempdir:
```

Comment:
this should be `if use_tmpfs and tmpfs_tempdir` - currently it always sets `METAFLOW_TEMPDIR`, breaking `metaflow.S3` when `tmpfs` is not enabled

Reply:
maybe it's easier to have one conditional at the top a'la
```
tmpfs_enabled = use_tmpfs or (tmpfs_size and use_tmpfs is None)
```
and have
```
if tmpfs_enabled and tmpfs_tempdir
```
here